infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationContext.java

```diff
@@ -15,21 +16,31 @@
 public class LogReplicationContext {
 
     @Getter
-    private LogReplicationConfig config;
+    private final LogReplicationConfigManager configManager;
 
     @Getter
-    @Setter
-    private TopologyDescriptor topology;
+    private final String localCorfuEndpoint;
 
     @Getter
-    private String localCorfuEndpoint;
+    @Setter
+    private long topologyConfigId;
+
 
     /**
      * Constructor
      **/
-    public LogReplicationContext(LogReplicationConfig config,  TopologyDescriptor topology, String localCorfuEndpoint) {
-        this.config = config;
-        this.topology = topology;
+    public LogReplicationContext(LogReplicationConfigManager configManager, long topologyConfigId, String localCorfuEndpoint) {
+        this.configManager = configManager;
+        this.topologyConfigId = topologyConfigId;
```

Comment:
after my PR for the upgrade manager is merged, it will also get initialized here, right?

Reply:
addressed offline